File: HackMateBackend/HackMate/src/main/java/com/example/HackMateBackend/dtos/notification/NotificationPreferencesDto.java
package com.example.HackMateBackend.dtos.notification;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class NotificationPreferencesDto {
    private boolean emailNotifications;
    private boolean pushNotifications;
    private boolean hackathonDeadlineReminders;
    private boolean joinRequestUpdates;
    private boolean teamUpdates;
    private boolean generalAnnouncements;
    private int reminderHoursBefore; // Hours before deadline to send reminder
}

File: HackmateFrontendFolder/app/src/main/java/com/example/hackmatefrontendfolder/domain/model/notification/NotificationPreferencesDto.kt
package com.example.hackmatefrontendfolder.domain.model.notification

data class NotificationPreferencesDto(

    val emailNotifications : Boolean,
    val pushNotifications : Boolean,
    val hackathonDeadlineReminders : Boolean,
    val joinRequestUpdates : Boolean,
    val teamUpdates : Boolean,
    val generalAnnouncements : Boolean,
    val reminderHoursBefore : Int

)
